src/privileged/trackers/api.js

```diff
@@ -54,6 +54,15 @@ this.trackers = class extends ExtensionAPI {
       context.extension.getURL("privileged/trackers/EveryWindow.js"));
     return {
       trackers: {
+        async checkExceptionStatus() {
+          const win = BrowserWindowTracker.getTopWindow({
+            private: false,
+            allowPopups: false,
+          });
+          const baseURI = win.document.documentURIObject;
```

Comment:
This is not the correct thing to check, this is the documentURI of the chrome window ("chrome://browser/content/browser.xul").

You _could_ use `win.gBrowser.currentURI` but that sounds flaky too, since it's not guaranteed that the unloading document is in the currently selected tab.

Ideally we would pass in a tabId and check with the contentPrincipal for that tab browser, but unfortunately I think when querying this after onbeforeunload we have no guarantee that the tab is still alive.

I would recommend taking a `hostname` argument and constructing a new `Services.io.newURI` with it, similar to how your test does it (might want to try-catch that). The hostname can be found on the data object in the beforeunload callback (even though we don't use it right now).

Does that make sense? :)

Reply:
yep!